src/Mesh.h

```diff
@@ -25,19 +25,43 @@ class Mesh
     public:
     Mesh(const std::string filename);
 
+    /**
+     * Get the physical x or y coordinate of an element node on the host Kokkos::View.
+     */
     [[nodiscard]]
-    inline double GetCoordinate(int element, int node, int dim) {return data_(element,node,dim+1);}
+    inline double GetCoordinate(int element, int node, int dim) const {return data_(element,node,dim+1);}
 
+    /**
+     * Get the global degree of freedom on the host Kokkos::View.
+     */
     [[nodiscard]]
-    inline int GetGlobalDof(int element, int node) {return data_(element,node,0);}
+    inline int GetGlobalDof(int element, int node) const {return data_(element,node,0);}
+
+    #ifdef KOKKOS_ENABLE_CUDA
+        /**
+         * Get the physical x or y coordinate of an element node on the device Kokkos::View.
+         */
+        [[nodiscard]] 
+        inline double GetDeviceCoordinate(int element, int node, int dim) const {return device_data_(element,node,dim+1);}
+        /**
+         * Get the global degree of freedom (DOF) number on the device Kokkos::View.
+         */
+        [[nodiscard]]
+        inline int GetDeviceGlobalDof(int element, int node) const {return device_data_(element,node,0);}
+    #endif
 
     private:
     size_t numVertices_ = 0;
     size_t numElements_ = 0;
     MeshType meshType_;
-    Kokkos::View<double***, Kokkos::DefaultHostExecutionSpace> data_; // CHANGE FROM HOST SPACE LATER
-
-    // data arrays depending on cuda flag
+    
+    // Mesh data in CPU memory space
+    Kokkos::View<double***, Kokkos::LayoutRight, Kokkos::HostSpace> data_; 
+    
+    // Mesh data in GPU memory space if using CUDA
+    #ifdef KOKKOS_ENABLE_CUDA
+        Kokkos::View<double***, Kokkos::LayoutRight, Kokkos::CudaSpace> device_data_; 
+    #endif
```

Comment:
Instead of using `data_` and `device_data_`, Let's replace them with `host_data_` and `data_` respectively. Then, with the `KOKKOS_ENABLE_CUDA` flag you can either deep copy or just do `data_ = host_data_` which will create a view without copying. Then the user can just use `data_`.